### DISCUSSION TITLE
[Script request] Network UPS Tools

### DISCUSSION THREAD

**Discussion by Hellac** [community-scripts/ProxmoxVE#69] [Request script] [answered]
### Application Name

Network UPS Tools

### Website

https://networkupstools.org/

### Description

The primary goal of the Network UPS Tools (NUT) project is to provide support for Power Devices, such as Uninterruptible Power Supplies, Power Distribution Units, Automatic Transfer Switches, Power Supply Units and Solar Controllers. NUT provides a common protocol and set of tools to monitor and manage such devices, and to consistently name equivalent features and data points, across a vast range of vendor-specific protocols and connection media types.

NUT provides many control and monitoring [features](https://networkupstools.org/features.html), with a uniform control and management interface. If you are just getting acquainted with NUT, [that page](https://networkupstools.org/features.html) also explains the technical design and some possible set-ups.

As of this publication, at least 184 different manufacturers, and 1193 hardware device models are known as [compatible](https://networkupstools.org/stable-hcl.html).

This software is the combined effort of many [individuals and companies](https://networkupstools.org/acknowledgements.html) with free and open source code licensed under the terms of GNU Public License (GPL), see [the NUT license text](https://github.com/networkupstools/nut/blob/master/COPYING) for specific details.

### Due Diligence

- [X] I have searched existing [scripts](https://community-scripts.github.io/Proxmox/scripts) and found no duplicates.
- [X] This is not a game-related request.
- [X] I have searched existing [discussions](https://github.com/community-scripts/ProxmoxVE/discussions?discussions_q=) and found no duplicate requests.

**Comment by DerMFalk** [comment, marked as answer]
What about this:
https://community-scripts.github.io/Proxmox/scripts?id=PeaNUT

**Reply by Hellac** [reply]
WHAT! I searched NUT, UPS, Network UPS Tools....

Thank you, I'll test this out and see how it goes.

**Comment by remz1337** [comment]
I have this in my setup. Not sure we need a script for this, it's installed through APT (`apt install nut`). Tricky part is passing through the UPS device (usb in my case). Then it's a matter of configuring the NUT server/client.

Plenty of tuto online
https://docs.deeztek.com/books/ubuntu/page/installing-nut-network-ups-tools-on-ubuntu-1804-lts

**Reply by hmoffatt** [reply]
Why not install it on the host instead? All the nut packages are in the standard Debian repositories, and there's no need to mess around with USB passthrough then. (And you'll want the whole node to shut down if the UPS is going offline too.)

**Comment by brianread108** [comment]
...and you can use the master/ slave structure to pass the UPS status to a VM or LXC using the network if you want to close down one properly in the event of an incident.

**Reply by hmoffatt** [reply]
Exactly, though you should get a clean shutdown of the VMs and containers if the whole host is shutting down anyway. 

I installed NUT on the host and connect to it from Home Assistant in an LXC to monitor the ups status.

**Reply by brianread108** [reply]
> Exactly, though you should get a clean shutdown of the VMs and containers if the whole host is shutting down anyway.
> 

For VMs only if you have the guest agent installed I suspect.

**Reply by hmoffatt** [reply]
I think it should issue an ACPI shutdown to the guest. Same as if you press the shutdown button at the console (or issue the equivalent proxmox shell command).

**Comment by Hellac** [comment]
Coming back to this, I looked at PeaNUT as DerMFalk suggested, unfortunately it is a dashboard rather than an assisted setup of NUT.

In my mind this script would help step a user through the setup of NUT:

- Check installation status of NUT
- identify and select UPS, set mode, assign users, notifications, emails, etc
- Assist in actions ups takes in event of power loss (immediately being shutdown? wait until <50% battery? etc)

I recognise that all of this is fairly straight forward, but having a script that steps a user through it all in plain english, and allows for changes to be made by rerunning the script to me feels like a worthwhile endeavor.

Appreciate all the discussion that has taken place!

**Reply by hmoffatt** [reply]
That sounds far beyond what the other helper scripts do, which is to install software rather than configure it.

**Reply by Hellac** [reply]
I would think this is installed directly on the PVE host, so would be similar in nature to the steps that occur when configuring an LXC in the other installation scripts. A series of prompts that update the required config files as needed.

**Reply by hmoffatt** [reply]
A general purpose configure script for NUT is not specific to Proxmox.

**Reply by VNRARA** [reply]
> A general purpose configure script for NUT is not specific to Proxmox.

Yet Processor Microcode and CPU scaling govenor scripts are also added. I believe a NUT-serverscript that would autofind a connected USB-device and setup the basics for NUT would fit right there as well.

**Reply by hmoffatt** [reply]
> Yet Processor Microcode and CPU scaling govenor scripts are also added. I believe a NUT-serverscript that would autofind a connected USB-device and setup the basics for NUT would fit right there as well.

It should still be delivered as part of NUT, not these scripts.
